persistence/azure-table/configuration_code_astp_[3,).partial.md

```diff
@@ -67,23 +67,21 @@ In case installers are enabled, but there's a need to opt out from creating the
 
 snippet: EnableInstallersConfigurationOptingOutFromTableCreation
 
-### Partitioning and compatibility helpers
+### Partitioning and compatibility mode helpers
 
-To atomically store data within the same transaction, all data written as part of the same message handling pipeline needs to share the same partition key.
-When storing sagas, saga IDs are by default deterministically derived from the saga data: the correlation property name and the correlation property value. Therefore, in conversations concerning a saga as part of the message handling pipeline, this deterministically derived ID is a perfect fit to be used as a partition key. Unfortunately, the saga ID assignment happens quite late during the message handling process and is not exposed to the user code.
+During a given message handling pipeline, multiple data storage operations may occur. To commit them atomically, in a single transaction, they must share a partition key. In conversations involving a saga, the saga ID is a good candidate for a partition key. Unfortunately, saga IDs are determined late in the message handling process and are not exposed to user code.
 
 This makes it difficult to:
 
-* Support transactional processing for existing sagas that were stored with the previous version of the persister and may have in-flight messages
-* Expose the deterministic saga ID early enough to enable business operations to participate in the transaction before the saga has been created
+* Enable transactional saga data storage for existing sagas that were stored with the previous version of the persister.
+* Allow other data operations to participate in the saga data transaction.
 
-To support the above scenarios the `IProvidePartitionKeyFromSagaId` interface implementation can be injected into behaviors in the logical pipeline stage.
+To support the above scenarios, `IProvidePartitionKeyFromSagaId` may be injected into behaviors at the logical pipeline stage:
 
 snippet: BehaviorUsingIProvidePartitionKeyFromSagaId
 
-This component enables users to:
+`IProvidePartitionKeyFromSagaId` enables users to set the partition key based on of the following:
 
-* Derive the table in which business- and saga data will be stored by looking for the `TableInformation` or falling back to the default table
```

Comment:
This is irrelevant here. While the snippet shows it, it is self explanatory, and it is _not_ something provided by `IProvidePartitionKeyFromSagaId`.

Reply:
This edit is incorrect. It is something provided by the helper tool

https://github.com/Particular/NServiceBus.Persistence.AzureStorage/blob/master/src/NServiceBus.Persistence.AzureTable/SagaPersisters/ProvidePartitionKeyFromSagaId.cs#L28-L36

Reply:
@danielmarbach I see, but I think the way this is laid out here is confusing. The component doesn't _allow users to derive the table_.... The component allows users to pass a value which may or may not be used as the partition key based on an algorithm, and part of that algorithm reads the table name. The confusion is exacerbated by the fact that the code snippet shows the user examining the table without using `IProvidePartitionKeyFromSagaId`:

```C#
if (context.Extensions.TryGet<TableInformation>(out var tableInformation))
{
    Log.Debug($"Table Information: {tableInformation.TableName}");
}
```

Perhaps it's best if we jump on a call next week and thrash this out quickly.

Reply:
@danielmarbach and I will work out an edit